#**LAB 16**

There are four problem statements given to you. Your task is to figure out what test to use. You are tasked to show the following:
1. State the statistical test you will use
2. Variables (IV and DV)
3. Hypotheses
4. Interpretation

Here are the libraries you would be needing.

T-Test:


```
import numpy as np
from scipy.stats import ttest_ind
```

Z-Test:


```
import scipy.stats as stats
import numpy as np
```

Chi-Square:


```
import numpy as np
from scipy.stats import chi2_contingency
```

Anova:


```
import numpy as np
from scipy.stats import f_oneway
```






##1. Problem Statement:
You are investigating whether there is a relationship between gender (Male/Female) and smartphone brand preference (Samsung, Apple, Other) among a sample of users. Conduct a test to determine if there is a significant association between gender and smartphone brand preference.

**State the statistical test you will use, Variables (IV and DV),
Hypotheses**


2 groups: gender/brand. Has categories.

Independent Variable: Gender

Dependent Variable: Smartphone brand preference

I will use chi-square test to determine a significant association between the two categorical variables. This is a test of independent variables.

Null(H0): There is no association between gender and smartphone brand preference as the variables are independent.

Alternative(H1): There is an association between gender and smartphone brand preference as the variables are not independent.

In [ ]:
# Import libraries
import numpy as np
from scipy.stats import chi2_contingency

In [ ]:
# Data
observed_data = np.array([[30, 20, 15], [25, 35, 10]])  # Example frequencies, rows represent gender, columns represent smartphone brand

In [ ]:
# Data
import pandas as pd
import numpy as np

observed_data = np.array([[30, 20, 15], [25, 35, 10]])

# Convert the array to a DataFrame
df = pd.DataFrame(observed_data, index=['Male', 'Female'], columns=['Brand A', 'Brand B', 'Brand C'])

# Display the DataFrame as a table
print(df)

        Brand A  Brand B  Brand C
Male         30       20       15
Female       25       35       10


In [ ]:
# Perform chi-squared test
chi2_stat, p_value, dof, expected = chi2_contingency(observed_data)

# Print the results
print(f"Chi-Squared Statistic: {chi2_stat}")
print(f"P-value: {p_value}")
print(f"Degrees of Freedom: {dof}")
print("Expected Frequencies:")
print(expected)

Chi-Squared Statistic: 5.367632367632368
P-value: 0.06830200317201793
Degrees of Freedom: 2
Expected Frequencies:
[[26.48148148 26.48148148 12.03703704]
 [28.51851852 28.51851852 12.96296296]]


**Interpretation:**

With a p-value of 0.068 > significance level (alpha) 0.05, we do not have enough evidence to reject the null(H0).

There is no significant association between gender and smartphone brand preference as the variables are independent.

##2. Problem Statement:
You are a data scientist analyzing the average response time (continuous) of two versions of a website feature. Conduct a test to determine if the average response time of Version A is significantly different from the expected average response time.

**State the statistical test you will use, Variables (IV and DV), Hypotheses**

Independent Variable is the website Version A and the expected average response time.

Dependent Variable is the average response time(continuous)

Null(H0): The average response time of Version A is not significantly different from the expected response time.

Alternative(H1): The average response time of Version A is significantly different from the expected response time.

We will need to compare the average response time of Version A to the expected average response time, I will use the one-sample t-test.

In [ ]:
import numpy as np
from scipy.stats import ttest_ind

In [ ]:
# Data
response_times_version_a = np.array([0.25, 0.28, 0.24, 0.26, 0.27])
expected_average_time = 0.3  # hypothetical value

In [ ]:
t_statistic, p_value = ttest_ind(response_times_version_a, expected_average_time)

In [ ]:
# Perform two-sample t-test
t_statistic, p_value = ttest_ind(response_times_version_a, expected_average_time)

# Print the results
print(f"Two-Sample T-Test Results:")
print(f"T-statistic: {t_statistic}")
print(f"P-value: {p_value}")

# Interpret the results
if p_value < 0.05:
    print("The means of the two groups are significantly different.")
else:
    print("There is no significant difference between the means of the two groups.")

Two-Sample T-Test Results:
T-statistic: -2.3094010767585003
P-value: 0.08208627983473407
There is no significant difference between the means of the two groups.


**Interpretation**

p-value 0.082 > significance level (alpha) 0.05, we fail to reject the null hypothesis.  

There is no significant difference between the average response time of Version A and the expected average response time. The average response time of Version A is not significantly different from the expected response time

##3. Problem Statement:
You have two datasets representing the scores of two groups of students (Group A and Group B) who underwent different teaching methods. Perform a test to determine if there is a significant difference in the mean scores of the two groups.

**State the statistical test you will use, Variables (IV and DV), Hypotheses**

2 group scores; 2 different teaching methods

Independent Variable: Teaching Method A and B

Dependent Variable: Test Scores

Null(H0): There is no significant difference in the mean scores of the two groups who underwent different teaching methods.

Alternative(H1): There is significant difference in the mean scores of the two groups who underwent different teaching methods.

I will use Paired Samples T-Test.

In [ ]:
# Import libraries
from scipy.stats import ttest_ind

In [ ]:
# Data
group_a_scores = np.array([85, 88, 92, 78, 95, 89, 91])
group_b_scores = np.array([76, 80, 82, 79, 88, 85, 87])

In [ ]:
# Perform two-sample t-test
t_statistic, p_value = ttest_ind(group_a_scores, group_b_scores)

# Print the results
print(f"Two-Sample T-Test Results:")
print(f"T-statistic: {t_statistic}")
print(f"P-value: {p_value}")

# Interpret the results
if p_value < 0.05:
    print("The means of the two groups are significantly different.")
else:
    print("There is no significant difference between the means of the two groups.")

Two-Sample T-Test Results:
T-statistic: 2.187379706640956
P-value: 0.04923856926406315
The means of the two groups are significantly different.


**Interpretation**

P-value 0.049 < significance level 0.05, we reject the null.  We reject there is no significant difference in the mean scores of the two groups who underwent different teaching methods.

##4. Problem Statement:
You have test scores from three different teaching methods (Method A, Method B, Method C). Conduct a one-way ANOVA to determine if there is a significant difference in the mean scores among the three teaching methods.

**State the statistical test you will use, Variables (IV and DV), Hypotheses**

Statistical Test: one-way ANOVA

Independent variable: 3 teaching methods

Dependent variable: scores

Do any of these three samples differ from each other?

Null(H0): There is no significant difference in the mean scores among the three teaching methods.

Alternative(H1): There is significant difference in the mean scores among the three teaching methods.

In [ ]:
# Import libraries
import numpy as np
from scipy.stats import f_oneway

In [ ]:
# Data
method_a_scores = np.array([80, 85, 88, 82, 87])
method_b_scores = np.array([75, 78, 80, 72, 79])
method_c_scores = np.array([88, 92, 85, 90, 89])

In [ ]:
# Perform one-way ANOVA
f_statistic, p_value = f_oneway(method_a_scores, method_b_scores, method_c_scores)

# Print the results
print(f"One-way ANOVA F-statistic: {f_statistic}")
print(f"P-value: {p_value}")

# Interpret the results
if p_value < 0.05:
    print("There are significant differences between the group means.")
else:
    print("There is no significant difference between the group means.")

One-way ANOVA F-statistic: 19.261324041811857
P-value: 0.0001795439780165966
There are significant differences between the group means.


**Interpretation**

The calculated F-statistic is 19.261 (greater than alpha 0.05). This measures the ratio of variance between the groups to the variance within the groups. A larger F-statistic suggests larger differences between the group means relative to the variability within the group.

P-value 0.0001 < significance level(alpha) 0.05.

We reject there is no significant difference in the mean scores among the three teaching methods.